daemon/src/setup_maker.rs

```diff
@@ -219,14 +219,20 @@ impl Actor {
 
 #[xtra_productivity(message_impl = false)]
 impl Actor {
-    fn handle(&mut self, msg: wire::SetupMsg) -> Result<()> {
-        let mut sender = self
-            .setup_msg_sender
-            .clone()
-            .context("Cannot forward message to contract setup task")?;
-        sender.send(msg).await?;
+    fn handle(&mut self, msg: wire::SetupMsg) {
+        let mut sender = match self.setup_msg_sender.clone() {
+            Some(sender) => sender,
+            None => {
+                tracing::error!(
+                    "Contract setup task not available, unable to dispatch contract setup message"
+                );
+                return;
+            }
+        };
 
-        Ok(())
+        if let Err(e) = sender.send(msg).await {
+            tracing::error!("Failed to dispatch contract setup message to task: {e}")
+        }
```

Comment:
🔧 To reduce the number of times you have to explicitly handle errors, you can extract the body of this method into a non-`xtra` method on `Actor` which returns a `Result`. Then you can handle that result here with a single `if let Err(e)`.

And then you can include the whole error stack with `{e:#}`.